python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint_search.py

```diff
@@ -0,0 +1,1113 @@
+                              "l2": system.getAtom(int(l2_idx)),
+                              "l3": system.getAtom(int(l3_idx))},
+            "equilibrium_values": {"r0": r0,
+                                   "thetaA0": thetaA0,
+                                   "thetaB0": thetaB0,
+                                   "phiA0": phiA0,
+                                   "phiB0": phiB0,
+                                   "phiC0": phiC0},
+            "force_constants": {"kr": kr,
+                                "kthetaA": kthetaA,
+                                "kthetaB": kthetaB,
+                                "kphiA": kphiA,
+                                "kphiB": kphiB,
+                                "kphiC": kphiC
+                                }}
+        # TODO: extract the best frame and feed it into Restraint
+        # Waiting for the BSS to fix the getFrames
+        # best_frame = traj.getFrames(index)
+        best_frame = system
+        restraint = Restraint(best_frame, restraint_dict,
+                              temperature,
+                              rest_type='Boresch')
+        return restraint
+
+    @staticmethod
+    def _boresch_restraint_BSS(u, system, temperature, lig_selection_str,
+                           recept_selection_str, work_dir, cutoff):
+        """Generate the Boresch Restraint.
+
+       Parameters
+       ----------
+
+       u : MDAnalysis.Universe
+           The trajectory for the ABFE restraint calculation as a
+           MDAnalysis.Universe object.
+
+       system : :class:`System <BioSimSpace._SireWrappers.System>`
+           The molecular system for the ABFE restraint calculation. This
+           must contain a single decoupled molecule and is assumed to have
+           already been equilibrated.
+
+       temperature : :class:`System <BioSimSpace.Types.Temperature>`
+           The temperature of the system
+
+       lig_selection_str: str
+           The selection string for the atoms in the ligand to consider
+           as potential anchor points.
+
+       recept_selection_str: str
+           The selection string for the protein in the ligand to consider
+           as potential anchor points.
+
+       work_dir : str
+           The working directory for the simulation.
+
+       cutoff: float
+           The greatest distance between ligand and receptor anchor atoms, in
+           Angstrom. Receptor anchors further than cutoff Angstroms from the closest
+           ligand anchors will not be included in the search for potential anchor points.
+
+       Returns
+       -------
+
+       restraint : :class:`Restraint <BioSimSpace.Sandpit.Exscientia.FreeEnergy.Restraint>`
+           The restraints of `rest_type` which best mimic the strongest receptor-ligand
+           interactions.
+        """
+        # Please don't review this part yet as Finlay is working on it.
+        # TODO Tidy this up and improve. Change algorithm to fit to match that discussed.
```

Comment:
Sorry, didn't see this bit. Will discuss with Finlay separately. Perhaps this can be done for his PR in to your devel?